napari/plugins/__init__.py

```diff
@@ -115,8 +115,12 @@ def set_call_order(self, new_order: CallOrderDict):
         for spec_name, hook_caller in self.hooks.items():
             order = []
             for p in new_order.get(spec_name, []):
+                try:
+                    imp = hook_caller.get_plugin_implementation(p['plugin'])
+                except KeyError:
+                    continue
+                imp.enabled = p['enabled']
```

Comment:
maybe i'm missing something really basic here, but if the try fails, then won't this line fail as `imp` isn't defined? maybe that's ok? maybe you could add a code comment too on why this is in a try/ except

Reply:
Because it fails, the `continue` is reached first so `imp.enabled = p['enabled']` never actually runs.

Reply:
ah and the continue moves it on :-) get it now! I do have a slight preference for the code in @goanpeca comment, but it was really just me missing the continue!!

Reply:
i guess what I prefer about the original implementation is that there's no question as to where the key might have occurred (was it `get_plugin_implementation` or was it `p['enabled']`?).

Reply:
ok, that's a fine rationale i hadn't considered that. fine to leave as is then